### PR TITLE
roboscape duplicated handlers improves #2361

### PR DESCRIPTION
**Pull request by hamidzr**
depends on https://github.com/NetsBlox/NetsBlox/pull/2360
- further reduces duplicated code on roboscape
- simplifies return values of `getRange` and `getTicks`

### CODE REVIEW

File: src/server/rpc/procedures/roboscape/roboscape.js [hunk 1]
```diff
@@ -153,18 +154,29 @@ RoboScape.prototype.getRobots = function () {
     return Object.keys(RoboScape.prototype._robots);
 };
 
+
+RoboScape.prototype._tbd = function (fnName, args) {
```
Comment (hamidzr) [inline src/server/rpc/procedures/roboscape/roboscape.js]: TODO: rename

File: src/server/rpc/procedures/roboscape/roboscape.js [hunk 2]
```diff
@@ -52,6 +52,7 @@ var RoboScape = function () {
 RoboScape.serviceName = 'RoboScape';
 RoboScape.prototype._robots = {};
 
+// fetch the robot, create one if necessary
 RoboScape.prototype._addRobot = function (mac_addr, ip4_addr, ip4_port) {
```
Comment (darrahts) [inline src/server/rpc/procedures/roboscape/roboscape.js]: I thought we were renaming this function?

Comment (hamidzr) [reply src/server/rpc/procedures/roboscape/roboscape.js]: good catch `getRobot` `getOrCreateRobot` ? any other suggestions

Comment (hamidzr) [reply src/server/rpc/procedures/roboscape/roboscape.js]: renamed to `getOrCreateRobot`

**Comment by hamidzr** [comment]
failed build is a flaky client test. reported to https://github.com/NetsBlox/Snap--Build-Your-Own-Blocks/issues/764

**Pull request merged**